src/core/js/models/questionModel.js

```diff
@@ -31,8 +31,30 @@ define([
         '_isSubmitted',
         '_score',
         '_isCorrect',
-        '_attemptsLeft'
+        '_attemptsLeft',
+        '_attemptStates'
       ]);
+    },
+
+    trackableType() {
+      return ComponentModel.resultExtend('trackableType', [
+        Boolean,
+        Number,
+        Boolean,
+        Number,
+        Array
+      ]);
+    },
+
+    setupModel(...args) {
+      super.setupModel(...args);
+      this.on('change:_isInteractionComplete', (model, value) => {
+        if (!value) {
```

Comment:
For clarity, suggest:
```suggestion
      this.on('change:_isInteractionComplete', (model, isInteractionComplete) => {
        if (!isInteractionComplete) {
```